Python/xOptionsMenu.py

```diff
@@ -1540,13 +1554,12 @@ def InitVideoControlsGUI(self):
 
     def _AspectRatio(self, w, h):
         """Returns the appropriate aspect ratio for the given resolution"""
-        ratios = ((5, 4), (4, 3), (3, 2), (16, 10), (5, 3), (16, 9), (16, 9.375),)
-        w = float(w) # comes in as string, want float (not int) for division
-        h = float(h)
-        for r in ratios:
-            # resolution is within 1 pixel wiggle room in any direction from the exact aspect ratio (needed to recognize 1280x854 as 3:2)
-            if (w+1)/(h-1) >= float(r[0])/float(r[1]) >= (w-1)/(h+1):
-                return " [%i:%i]" % (r[0], r[1])
+        # allow some wiggle room for float comparisons (CWalter particularly points out 1280x854)
```

Comment:
Nitpicks:
• I prefer to have my name spelled with an h.
• The wiggle room wasn’t primarily for float comparison (although that was a welcome side effect), but exactly for matching aspect ratios that aren’t mathematically precise (854*3/2 = 1281, not 1280). Obviously you have now found ones that are even further off than what I tested at the time. (So maybe there’s no need for dealing with pixels anymore and one could just accept a certain range of ratios.)

Reply:
Whoops! Sorry about the typo in your name--I will fix that.

I do realize why you had the wiggle room -- to me, float comparison and mathematically imprecise resolutions are symptoms of the same problem. It seemed to me that aspect ratios at most were expressed to the hundredths place in various online fora, so I went with that much precision.

I did run some calculations and saw some edge cases that the wiggle room would catch that my stupid rounding would not, however. When they both act together, all the resolutions offered on my ultrawide display have an aspect ratio listed. I suppose I could delete the wiggle room and see if that causes any issues.